File: 3788-maximum-unique-subarray-sum-after-deletion/3788-maximum-unique-subarray-sum-after-deletion.cpp
class Solution {
public:
    int maxSum(vector<int>& nums) {
        unordered_set<int> inArr;
        int res = -1;
        int sum = 0;
        int maxNeg = INT_MIN;
        for(int i = 0; i < nums.size(); i++)
        {
            if(nums[i] >= 0 && !inArr.count(nums[i]))
            {
                sum += nums[i];
                res = max(res, sum);
                inArr.insert(nums[i]);
            }
            else if(nums[i] < 0)
            {
                maxNeg = max(maxNeg, nums[i]);
            }
        }
        if(res == -1)
        {
            return maxNeg;
        }
        else
        {
            return res;
        }
    }
};

File: 3788-maximum-unique-subarray-sum-after-deletion/3788-maximum-unique-subarray-sum-after-deletion.py
class Solution:
    def maxSum(self, nums: List[int]) -> int:
        inArr = set()
        res = -1
        sum = 0
        maxNeg = float("-inf")
        for num in nums:
            if num >= 0 and num not in inArr:
                sum += num
                res = max(res, sum)
                inArr.add(num)
            else:
                maxNeg = max(maxNeg, num)
        if res == -1:
            return maxNeg
        else:
            return res
        